number/src/serialize.rs

```diff
@@ -102,24 +103,19 @@ pub fn buffered_write_file<R>(
     Ok(result)
 }
 
-pub fn write_polys_file<F: FieldElement>(
-    path: &Path,
-    polys: &[(String, Vec<F>)],
-) -> Result<(), serde_cbor::Error> {
-    buffered_write_file(path, |writer| write_polys_stream(writer, polys))??;
-
-    Ok(())
-}
-
-fn write_polys_stream<T: FieldElement>(
-    file: &mut impl Write,
-    polys: &[(String, Vec<T>)],
-) -> Result<(), serde_cbor::Error> {
-    serde_cbor::to_writer(file, &polys)
+pub trait ReadWrite {
+    fn read(file: &mut impl Read) -> Self;
+    fn write(&self, path: &Path) -> Result<(), serde_cbor::Error>;
 }
 
-pub fn read_polys_file<T: FieldElement>(file: &mut impl Read) -> Vec<(String, Vec<T>)> {
-    serde_cbor::from_reader(file).unwrap()
+impl<T: DeserializeOwned + Serialize> ReadWrite for T {
+    fn read(file: &mut impl Read) -> Self {
+        serde_cbor::from_reader(file).unwrap()
```

Comment:
unrelated to this PR but we should probably use `BufRead` here

Reply:
Do you mean just changing to `file: &mut impl BufRead`? What would be the difference? It is just a trait that adds more functions, which apparently `serde_cbor::from_reader()` does not use?

Reply:
The difference would be that if you try to pass a simple `File`, it would error out at compile time, which is good, since reading a file line by line is really inefficient. It's unrelated to this PR so feel free to close.

Reply:
Yeah, think it should be in a different PR, but thanks for explaining, wasn't aware of that!